gobblin-core/src/main/java/org/apache/gobblin/source/extractor/filebased/FileBasedSource.java

```diff
@@ -215,10 +215,11 @@ protected void initLogger(SourceState state) {
       results = this.fsHelper.ls(path);
       for (int i = 0; i < results.size(); i++) {
         URI uri = new URI(results.get(i));
-        File file = uri.isAbsolute()?
-            new File(uri) : new File(state.getProp(ConfigurationKeys.SOURCE_FILEBASED_DATA_DIRECTORY), uri.toString());
-
-        String filePath = file.getAbsolutePath();
+        String filePath = uri.toString();
```

Comment:
What's the reason for this change?

Reply:
The previous implementation has an bug. It doesn't work for absolute uri with hdfs schema because java `File()` only represents local file, expecting schema `file://`